contracts/base/ModuleManager.sol

```diff
@@ -40,6 +40,29 @@ abstract contract ModuleManager is SelfAuthorized, Executor, GuardManager {
         }
     }
 
+    function runPreExecutionChecks(
+        address to,
+        uint256 value,
+        bytes memory data,
+        Enum.Operation operation,
+        address guard
+    ) internal returns (bytes32 guardHash) {
+        // Only whitelisted modules are allowed.
+        require(msg.sender != SENTINEL_MODULES && modules[msg.sender] != address(0), "GS104");
+
+        if (guard != address(0)) {
+            guardHash = Guard(guard).checkModuleTransaction(to, value, data, operation, msg.sender);
+        }
+    }
+
+    function postExecutionChecksWithEventEmissions(bytes32 guardHash, bool success, address guard) internal {
+        if (guard != address(0)) {
+            Guard(guard).checkAfterExecution(guardHash, success);
+        }
+        if (success) emit ExecutionFromModuleSuccess(msg.sender);
+        else emit ExecutionFromModuleFailure(msg.sender);
+    }
```

Comment:
would be good to add natspec here. Everything else looks good to me

Reply:
Done 👍🏾